### ISSUE TITLE
Question regarding reimplementation abnormal results

### ISSUE THREAD

**Issue by shuchenYee** [MaxSobolMark/PolicyAgnosticRL#14]
Hi authors! I'm trying to build upon it in my own research about fine-tuning diffusion policy, which is trained by Torch. So I am currently re-implementing your method in PyTorch. 

I have been focusing on the offline training stage, but I'm running into some results that I'm not sure how to interpret, and I was hoping you might have some ideas.

My core question is whether my actor network is converging correctly. In addition, the reward and loss curves look quite abnormal to me, and I'm struggling to diagnose the potential cause.

Here is a summary of my setup for the offline training phase:

1. Dataset: I'm using the same dataset that was used to pre-train the diffusion policy.

2. Reward Model: I've implemented a simple reward function. For each episode, I construct (state, action, next_state, reward, dones) transitions. The reward for the transition from t to t+1 is defined as the negative L2 distance to the goal:

`r_t = -np.linalg.norm(pos_t1 - goal_pos)`

Here, pos_goal is the end-effector position from the final frame of that specific episode.

3. Done Signal: The done flag is set to `True` only for the very last transition in each episode. For all other transitions, it is `False`.

Given this setup, I'm observing strange learning curves (please see attached plots below). I was wondering if you might have any initial thoughts on what could be going wrong, or if this behavior is perhaps expected under certain conditions based on your experience.

<img width="741" height="709" alt="Image" src="https://github.com/user-attachments/assets/8b3cdf71-bf8c-4fb1-874d-823342861673" />
<img width="1091" height="672" alt="Image" src="https://github.com/user-attachments/assets/1656cb75-69ab-4477-896f-fe897aee0ab0" />
<img width="1101" height="676" alt="Image" src="https://github.com/user-attachments/assets/895a4a8b-2acc-42b3-b8c0-bfcc40517569" />
<img width="1104" height="678" alt="Image" src="https://github.com/user-attachments/assets/32971538-9127-425a-9db1-b869b9efb410" />

Any insight you could provide would be greatly appreciated.

Thank you for your time.
